assets/course-theme/blocks/template-style/template-style.block.json

```diff
@@ -0,0 +1,13 @@
+{
+	"name": "sensei-lms/template-style",
+	"category": "theme",
+	"parent": [],
+	"apiVersion": 2,
+	"attributes": {
```

Comment:
I know it's the purpose, but I'm still a bit worried we'll get stuck with CSS in user sites that we can't fix. 

One thing to prepare for that is that we could add the template version to this block as an attribute. Maybe then we'll want to update the styles with minor or patch-level changes.

This attribute is also a better place for the version than the current approach of adding a class somewhere in the HTML that I already forgot to do for the new templates :)

Reply:
If we add the css version here then we'll have to support a css version in this "Template Style" block attribute and also the legacy template/css versioning, which is adding a class somewhere in the HTML.

I personally think there should be only two versions of each template+css pair. One is the latest version that we ship and the other is the user's customized version of the template/css pair. If we think there is a css that might affect LM templates when changed, then that css should be included into block template content.

Otherwise we will end up with a large directory of html+css versions for each lm template because we need to version each time we make a small adjustment.

Reply:
This would replace the version tag added as a class to the HTML. Though we can still use that instead, it's just a bit more fragile since that class could be removed by the user.

We don't need to keep older versions of the HTML nor the CSS in the codebase, all CSS that depends on the template HTML would be included as template CSS, and thus saved in the database with this block. (Unless we do a major change like we are doing now.)